core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -1106,6 +1159,26 @@ else if (isInferenceCandidate(conjunct)) {
                     TRUE);
         }
 
+        private static boolean isSimpleExpression(Expression expression)
+        {
+            return isSimpleExpression(expression, true);
+        }
+
+        private static boolean isSimpleExpression(Expression expression, boolean allowArithmeticBinaryExpression)
```

Comment:
Could this be implemented by deriving from DefaultTraversalVisitor ? e.g. DeterminismEvaluator#isDeterministic

Reply:
It could. It would be way more verbose and harder to read. The recursion here is very shallow btw.